src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -533,10 +533,13 @@ public void SavePackageSources(IEnumerable<PackageSource> sources)
             // add entries to the disablePackageSource for disabled package sources that are not in loaded 'sources'
             foreach (var setting in existingDisabledSources)
             {
-                if (!sourcesToDisable.Any(
-                    s => s.Key == setting.Key
-                    && s.Priority == setting.Priority
-                    && s.IsMachineWide == setting.IsMachineWide))
+                // The following code ensures that we do not miss to mark an existing disabled source as disabled.
```

Comment:
Just updated the PR. I have added a passing unit test to cover this scenario. Running a full build now on my machine